File: app/src/main/java/com/example/javabutton/AoiJavaboWidgetProvider.java
package com.example.javabutton;

/**
 * Created by davy on 15/09/18.
 */
public class AoiJavaboWidgetProvider extends JavaButtonWidgetProvider {
	@Override
	protected String getJavaActionName() { return "com.example.javabutton.javabuttonwidget.AOI_ACTION"; }
	@Override
	protected int getJavaboLayoutId() { return R.layout.aoi_appwidget; }
	@Override
	protected int getJavaBotanId() { return R.id.aoiButton; }
}

File: app/src/main/java/com/example/javabutton/AoiJavaboWidgetProvider.kt
package com.example.javabutton

class AoiJavaboWidgetProvider : JavaButtonWidgetProvider() {
    protected override val javaActionName: String
        get() =  "com.example.javabutton.javabuttonwidget.AOI_ACTION"

    protected override val javaboLayoutId: Int
        get() = R.layout.aoi_appwidget

    protected override val javaBotanId: Int
        get() = R.id.aoiButton
}
